Cabal/Distribution/Simple/Setup.hs

```diff
@@ -256,7 +258,7 @@ data ConfigFlags = ConfigFlags {
     configStripLibs :: Flag Bool,      -- ^Enable library stripping
     configConstraints :: [Dependency], -- ^Additional constraints for
                                        -- dependencies.
-    configDependencies :: [(PackageName, ComponentId)],
+    configDependencies :: [(PackageName, ComponentName, ComponentId)],
```

Comment:
This is now a 3-tuple. Should it be a proper datatype? something like `data GivenDependency` or `GivenComponent`

Reply:
I'd say yes, if it's not too much of a trouble.